src/spaceObjects/playerSpaceship.cpp

```diff
@@ -1620,6 +1645,20 @@ void PlayerSpaceship::onReceiveClientCommand(int32_t client_id, sf::Packet& pack
     case CMD_SET_SCIENCE_LINK:
         {
             packet >> linked_science_probe_id;
+
+            if (linked_science_probe_id != -1 && on_probe_link.isSet())
+            {
+                P<ScanProbe> p = game_server->getObjectById(linked_science_probe_id);
+
+                if (p)
+                {
+                    on_probe_link.call(P<PlayerSpaceship>(this), P<ScanProbe>(p));
+                }
+            }
+            else if (linked_science_probe_id == -1 && on_probe_unlink.isSet())
+            {
+                on_probe_unlink.call(P<PlayerSpaceship>(this));
```

Comment:
We could give the (old) probe here as 2nd parameter, might be nice to keep it in parallel with linking?